src/librustdoc/core.rs

```diff
@@ -257,7 +257,10 @@ where
         .filter_map(|lint| {
             // Permit feature-gated lints to avoid feature errors when trying to
             // allow all lints.
-            if lint.name == warnings_lint_name || lint.feature_gate.is_some() {
+            if lint.name == warnings_lint_name
+                || lint.feature_gate.is_some()
+                || allowed_lints.iter().any(|l| lint.name == l)
```

Comment:
Now that you're doing this check, I wonder if we still need to mark the "warnings" lint as "allowed" (meaning, not caped to "allowed" haha). Can you give it a try please?

Reply:
Good idea, changed and all the rustdoc-ui tests pass locally :)

Reply:
Yeay! \o/